components/mpas-ocean/src/shared/mpas_ocn_scaled_sfwf.F

```diff
@@ -0,0 +1,567 @@
+!
+!  routine ocn_scaled_sfwf_init
+!
+!> \brief   Initialize scaling of data freshwater fluxes
+!> \author Shixuan Zhang 
+!> \date   May 2025
+!> \details
+!>  Set alarms needed to compute yearly and running means of water fluxes 
+!
+!-----------------------------------------------------------------------
+
+    subroutine ocn_scaled_sfwf_init(domain)!{{{
+
+        type (domain_type), intent(inout) :: domain
+
+        ! Alarm variables
+        type (mpas_Time_Type) :: alarmTime
+        type (mpas_TimeInterval_type) :: alarmTimeStep
+
+        type (block_type), pointer :: block_ptr
+        type (mpas_pool_type), pointer :: meshPool
+
+        real (kind=RKIND), pointer :: sphere_radius
+
+        integer :: err_tmp
+
+        character(len=19) :: timeString
+
+        if (config_use_precip_scaling) then
+            scaledSFWFOn = .true.
+        else
+            scaledSFWFOn = .false.
+            return
+        endif
+
+        if (trim(config_precip_scaling_mode) /= 'time-dependent' .and.  &
+            trim(config_precip_scaling_mode) /= 'constant') then
+            call mpas_log_write('scaledSFWFOn = .true. requires config_precip_scaling_mode = time-dependent or constant', &
+                                mpas_LOG_CRIT)
+        end if
+
+        if (config_precip_scaling_history_years < 1) then
+            call mpas_log_write('scaledSFWFOn = .true. requires config_precip_scaling_history_years >= 1', &
+                                mpas_LOG_CRIT)
+        endif
+        
+        ! Setup time interval string based on config_precip_scaling_history_years (treated as "years")
+        write(timeString, '( I4.4,"-00-00_00:00:00" )') config_precip_scaling_history_years
+
+        ! Setup Alarm for updating of scaling factors for precipitation flux
+        alarmTime = mpas_get_clock_time(domain % clock, &
+                                        mpas_START_TIME, &
+                                        ierr=err_tmp)
+        
+        call mpas_set_timeInterval(alarmTimeStep, &
+                                   timeString=timeString,& 
+                                   ierr=err_tmp)
+
+        call mpas_add_clock_alarm(domain % clock, alarmID, alarmTime + alarmTimeStep, &
+                                  alarmTimeInterval=alarmTimeStep, ierr=err_tmp)
+        ! Density of sea water
+        rhow = rho_sw
+
+        ! Compute Earth's surface area (used for normalizing fluxes)
+        if (config_precip_scaling_write_to_logfile) then
+           block_ptr => domain % blocklist
+           do while(associated(block_ptr))
+              call mpas_pool_get_subpool(block_ptr % structs, 'mesh', meshPool)
+
+              ! Extract earth radius 
+              call mpas_pool_get_config(meshPool, 'sphere_radius', sphere_radius)
+
+              ! Compute Earth's surface area (used for normalizing fluxes)
+              earthArea = 4.0_RKIND * pi * sphere_radius * sphere_radius
```

Comment:
```suggestion
              ! Compute Earth's surface area (used for normalizing fluxes)
              earthArea = 4.0_RKIND * pi * sphere_radius * sphereRadius
```

Since you have `use ocn_mesh` above , and `sphereRadius` is a public variable in `ocn_mesh`, you can make this change
https://github.com/E3SM-Project/E3SM/blob/d3ac406ece15206a4e2391f761ebd6912898b9c9/components/mpas-ocean/src/shared/mpas_ocn_mesh.F#L51-L52

Reply:
Thank you @cbegeman: this is great! I've made the changes according to your suggestion.